eth/api.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2015 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package eth
+
+import (
+	"errors"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/ethash"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/compiler"
+	"github.com/ethereum/go-ethereum/logger"
+	"github.com/ethereum/go-ethereum/logger/glog"
+	rpc "github.com/ethereum/go-ethereum/rpc/v2"
+)
+
+// PublicEthereumApi provides an API to access Ethereum related information.
+// It offers only methods that operate on public data that is freely available to anyone.
+type PublicEthereumApi struct {
```

Comment:
Not sure any of the methods in this file should be in eth  except for `GasPrice`, `Syncing` and `ProtocolVersion`, the rest I suggest go to where they belong:

* hashrate, (auto)DAG and miner related stuff should be in miner
* coinbase, etherbase, listAccount, newAccount, unlockAccount, lockAccount should be in accounts
* solc related stuff in common/compiler

The ethereum service (backend.go) could insert these modules into Apis() but probably even that should be in the wrapper (cmd/geth/main or cmd/utils/flags).